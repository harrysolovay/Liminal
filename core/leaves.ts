import { Ty } from "./Ty.ts"

export type none<P extends string = string> = Ty<null, P>
export const none: Ty<null, never> = Ty((description) => ({
  type: "null",
  description,
}))

export type bool<P extends string = string> = Ty<boolean, P>
export const bool: Ty<boolean, never> = Ty((description) => ({
  type: "boolean",
  description,
}))

export type num<P extends string = string> = Ty<number, P>
export const num: Ty<number, never> = Ty((description) => ({
  type: "number",
  description,
}))

export type str<P extends string = string> = Ty<string, P>
export const str: Ty<string, never> = Ty((description) => ({
  type: "string",
  description,
}))
