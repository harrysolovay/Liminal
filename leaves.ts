import { Ty } from "./Ty.ts"

export const none = Ty<null>((description) => ({
  type: "null",
  description,
}))

export const boolean = Ty<boolean>((description) => ({
  type: "boolean",
  description,
}))

export const number = Ty<number>((description) => ({
  type: "number",
  description,
}))

export const string = Ty<string>((description) => ({
  type: "string",
  description,
}))
