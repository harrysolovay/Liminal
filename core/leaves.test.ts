import { bool, none, num, str } from "./leaves.ts"
import { assertTySnapshot } from "../test_util.ts"

Deno.test("bool", async (t) => {
  await assertTySnapshot(t, bool)
})

Deno.test("none", async (t) => {
  await assertTySnapshot(t, none)
})

Deno.test("num", async (t) => {
  await assertTySnapshot(t, num)
})

Deno.test("str", async (t) => {
  await assertTySnapshot(t, str)
})
