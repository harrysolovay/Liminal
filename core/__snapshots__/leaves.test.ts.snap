export const snapshot = {};

snapshot[`bool 1`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "",
      type: "boolean",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`bool 2`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description one.",
      type: "boolean",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`bool 3`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description two. Description one.",
      type: "boolean",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`none 1`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "",
      type: "null",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`none 2`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description one.",
      type: "null",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`none 3`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description two. Description one.",
      type: "null",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`num 1`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "",
      type: "number",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`num 2`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description one.",
      type: "number",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`num 3`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description two. Description one.",
      type: "number",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`str 1`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "",
      type: "string",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`str 2`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description one.",
      type: "string",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`str 3`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      description: "Description two. Description one.",
      type: "string",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;
